### DISCUSSION TITLE
Getting error when converting PNG - Expected 4 bytes; found 2 bytes

### DISCUSSION THREAD

**Discussion by juancxt** [ImageMagick/ImageMagick#2244] [Help] [answered]
The attached signature PNG file throws a "Expected 4 bytes; found 2 bytes" exception when trying to convert or when using the identify application.  I was hoping that someone could take a look at the PNG file and let me know what is wrong with it.

![signature](https://user-images.githubusercontent.com/56411435/86772161-21ae2e00-c019-11ea-941f-cc35a3122e18.png)

**Comment by snibgo** [comment, marked as answer]
exiftool says "Warning : Truncated PNG IEND chunk".

pngcheck says:
```
zlib warning:  different version (expected 1.2.8, using 1.2.11)

sign.png  EOF while reading CRC value
ERROR: sign.png
```

Gimp 2.10.0 mis-reads the file, and thinks the image is entirely transparent black.

Microsoft Photo Viewer can read the file, and re-save it as a PNG, but that gives the same error.

Microsoft mspaint can read the file, and re-save it as a PNG without transparency, and that PNG has no problems.

**Comment by juancxt** [comment]
snibgo - Thank you!!!  The file is missing the last two bytes.  I was able to append the missing bytes and the file converts without errors.  Now to convince the user to produce files that aren't borked.  :-(
